### PR TITLE
dev warnings for missing reducer or middleware

### PR DESCRIPTION
**Pull request by phryneas**

### CODE REVIEW

File: src/query/tests/devWarnings.test.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    })
+    store.dispatch(api1.endpoints.q1.initiate(undefined))
+    store.dispatch(api2.endpoints.q1.initiate(undefined))
+    expect(getLog().log)
+      .toBe(`Warning: Middleware for RTK-Query API at reducerPath "api1" has not been added to the store.
+Features like automatic cache collection, automatic refetching etc. will not be available.
+Warning: Middleware for RTK-Query API at reducerPath "api2" has not been added to the store.
+Features like automatic cache collection, automatic refetching etc. will not be available.`)
+  })
+})
```
Comment (msutkowski) [inline src/query/tests/devWarnings.test.tsx:86]: How do you feel about an assertion for not logging with `NODE_ENV='production'` for completeness? And the same for the reducer test below?

File: src/query/tests/helpers.tsx [hunk 2]
```diff
@@ -75,31 +74,37 @@ export const useRenderCounter = () => {
...
-) {
-  const actions = _actions.concat()
-  actions.shift() // remove INIT
-  expect(matchers.length).toBe(actions.length)
-  for (let i = 0; i < matchers.length; i++) {
-    expect(matchers[i](actions[i])).toBe(true)
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toMatchSequence(...matchers: Array<(arg: any) => boolean>): R
```
Comment (msutkowski) [inline src/query/tests/helpers.tsx:80]: :fire:

File: src/query/tests/helpers.tsx [hunk 3]
```diff
@@ -75,31 +74,37 @@ export const useRenderCounter = () => {
...
+    actions.shift() // remove INIT
+
+    for (let i = 0; i < matchers.length; i++) {
+      if (!matchers[i](actions[i])) {
+        return {
+          message: () =>
+            `Action ${actions[i].type} does not match sequence at position ${i}.`,
+          pass: false,
+        }
+      }
```
Comment (msutkowski) [inline src/query/tests/helpers.tsx:100]: :fire: yes please.

**Review by msutkowski — COMMENTED** [review]
Awesome! LGTM, but I'd just recommend we assert the production behavior as well on the warnings.

File: src/query/tests/devWarnings.test.tsx (hunk 1 above)
Comment (phryneas) [reply src/query/tests/devWarnings.test.tsx]: happy? :)

**Review by msutkowski — APPROVED** [review]
:trophy: :rocket:

File: src/query/tests/devWarnings.test.tsx (hunk 1 above)
Comment (msutkowski) [reply src/query/tests/devWarnings.test.tsx]: Very, thanks!

**Pull request merged**